subt/main.py

```diff
@@ -314,6 +316,10 @@ def follow_wall(self, radius, right_wall=False, timeout=timedelta(hours=3), dist
                     reason = REASON_VIRTUAL_BUMPER
                     break
 
+                loop = self.loop_detector.loop()
```

Comment:
What is the relation between how often we run through this loop and how often we update the detector in on_pose3d()? This function is relatively "heavy" to compute, so we should call it as rarely as possible. Alternatively, the loop detector could cache (and invalidate on update) the output of the loop() call.

Reply:
`on_pose3d` is called when localization announces new pose (the link is `["fromrospy.pose3d", "app.pose3d"]`). I am not sure how often that is.

Reply:
the IMU has crazy update (250Hz?) so the `pose3d` may be as bad

Reply:
As discussed over e-mail, pose3d turns out to be much (90x) rarer.

Reply:
I see now that this part is called with **every** message, i.e. not good place to detect loops (although there is now cache for loop detection). OTOH this is the place where it acts ... so I am not sure if to limit it to `if channel=="pose3d"`. Thanks m.

Reply:
So... do we want to change anything about this line now or not?